webapp-django/crashstats/crashstats/static/crashstats/js/socorro/oauth2.js

```diff
@@ -110,8 +121,18 @@ var OAuth2 = (function() {
                         $.post(url, data)
                         .done(function(response) {
                             // It worked!
-                            // TODO: https://bugzilla.mozilla.org/show_bug.cgi?id=1283296
-                            document.location.reload();
+                            var next = getParameterByName('next');
+                            // only if ?next=/... exists on the current URL
+                            if (next) {
+                                // A specific URL exits.
+                                // This is most likely the case when you tried
+                                // to access a privileged URL whilst being
+                                // anonymous and being redirected.
+                                // Make sure it's on this server
+                                document.location.pathname = next;
+                            } else {
```

Comment:
I've convinced myself that `searchParams` is a decently safe way to do this, if a bit nerve wracking. It only works in Firefox though. I don't know if that's a concern for you. I'd put some sort of fallback, in case of other browsers. At least a `return` afterwards, in case the browser falls through.